op-e2e/external_geth/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/external"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+)
+
+func main() {
+	var init bool
+	var configPath string
+	flag.BoolVar(&init, "init", false, "Do one time setup for all executions")
+	flag.StringVar(&configPath, "config", "", "Execute based on the config in this file")
+	flag.Parse()
+	err := run(init, configPath)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func build() error {
+	cmd := exec.Command("go", "build", "-o", "op-geth", "github.com/ethereum/go-ethereum/cmd/geth")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
```

Comment:
I don't think I'd include the building logic here, but just make it a requirement that the shim and client are built before the tests are executed. You can usually get a command to execute before a test runs fairly easily  - especially with make but even in most IDEs.

Reply:
I discussed this a little more in my response to the `TestMain`.  Doing golang development, I generally expect to be able to run `go test` and have things work.  Actually, many of the recent changes forcing `make devnet-allocs` to be a prerequisite and requiring `make` to be used to invoke the tests feels like a significant regression to me, especially considering that all of these tests used to run just fine with simple `go test` invocations on the command line.

To me, doing the build from the shim is nicer, as it ensures that whether using IDE, `go test`, or `make`, that your test invocation will work with no additional setup or configuration.  I'll go ahead and move the external client build to `make`, but I'm unconvinced it's the right thing to do.

Reply:
Yeah I hear your concerns. I just think that `go test` is too low level a tool once you get into the realm of end-to-end tests. We're effectively using `make` as the canonical entry point as its that bit higher level and gives us better control over dependencies and the overall build pipeline. In an ideal world we should be testing the already built binaries - a build pipeline should build once and then the rest of the pipeline tries to find fault in that build.  We're a fair way off that right now but having e2e tests build their own dependencies moves us even further away from it.

Reply:
I think this might be the slight schizophrenia I feel with the direction and goals of the e2e.

In our initial discussions there was a strong desire to keep the in-process op-geth because of the ease of iterative development it facilitates.  But, necessarily by testing op-geth by compiling it into the test binary created by `go test`, we're preventing these tests from truly executing against a release.  The same could be said for the other assorted components.

If the goal of the op-e2e is to execute against release binaries built elsewhere, then I totally agree that `make` is a fine tool to ensure they're there (though you'll have trouble convincing this gopher that `go test` isn't also fine e2e tool).  That just feels in conflict with those other discussions. 

But, regardless of the build implementation, I _do_ feel like allowing external binaries to be leveraged in the e2e is valuable -- both for op-geth itself and for other clients wishing to leverage these tests -- so I'd love to see this merged.